From 953d55f7b88add7ff5e132603d0c23372ece12a8 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 24 Nov 2022 11:22:00 +0200
Subject: [PATCH] [SW-105079] gaudi3: clear boot termination

In secure boot mode, for non-cold resets the access from PCIe to
NoC is blocked by HW FSM. Due to this, any access from PCIe I/F
to ASIC will be blocked. FW need to clear boot termination in order
to un-block the access. This patch add changes to clear boot
termination as part of hl_pcie_enable_access function. This configuration
also needs to be done in PCIe link up sequence in ppboot and erom.

Change-Id: I43cb7874cf3722917312d9d97d7f12f910195cbd
---

diff --git a/arch/gaudi3/common/common.c b/arch/gaudi3/common/common.c
index b5b026e..94a9a66 100644
--- a/arch/gaudi3/common/common.c
+++ b/arch/gaudi3/common/common.c
@@ -8,12 +8,20 @@
 #include <d0_pcie_wrap_regs.h>
 #include <d0_pcie_core_regs.h>
 #include <d0_pcie_aux_regs.h>
+#include <d0_pcie_aux_masks.h>
 #include <d0_pcie_elbi_rr_mstr_if_rr_lbw_regs.h>
+
+void hl_pcie_clear_boot_termination(void)
+{
+	regmap_writel(D0_PCIE_AUX_BOOT_TERM_CLR_CLR_M, mmD0_PCIE_AUX_BOOT_TERM_CLR);
+}
 
 void hl_pcie_enable_access(void)
 {
 	unsigned int rst_src;
 
+	hl_pcie_clear_boot_termination();
+
 	/* FSM blocks all range hence openning here */
 	regmap_writel(0, mmD0_PCIE_ELBI_RR_MSTR_IF_RR_LBW_PRIV_RANGE_EN_SHORT_13);
 	rst_src = hl_rst_src_get();
